frontend/src/screens/wallet/send/Onchain.tsx

```diff
@@ -0,0 +1,234 @@
+        }
+        navigate(`/wallet/swap/out/status/${swapOutResponse.swapId}`);
+        toast({ title: "Initiated swap" });
+      } else {
+        const response = await request<RedeemOnchainFundsResponse>(
+          "/api/wallet/redeem-onchain-funds",
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              toAddress: address,
+              amount: +amount,
+              feeRate: +feeRate,
+            }),
+          }
+        );
+        if (!response?.txId) {
+          throw new Error("No address in response");
+        }
+        navigate(`/wallet/send/onchain-success`, {
+          state: {
+            amount: +amount,
+            txId: response.txId,
+          },
+        });
+        toast({
+          title: "Successfully broadcasted transaction",
+        });
+      }
+    } catch (e) {
+      toast({
+        variant: "destructive",
+        title: "Failed to send payment",
+        description: "" + e,
+      });
+      console.error(e);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  React.useEffect(() => {
+    if (!address) {
+      navigate("/wallet/send");
+    }
+  }, [navigate, address]);
+
+  if (!balances || !address || (!recommendedFees && !mempoolError)) {
+    return <Loading />;
+  }
+
+  return (
+    <form onSubmit={onSubmit} className="grid gap-6 md:max-w-lg">
+      <div className="grid gap-2">
+        <Label>Recipient</Label>
+        <div className="flex flex-wrap gap-2 items-center">
+          {address.match(/.{1,4}/g)?.map((word, index) => {
+            if (index % 2 === 0) {
+              return (
+                <span key={index} className="text-foreground">
+                  {word}
+                </span>
+              );
+            } else {
+              return (
+                <span key={index} className="text-muted-foreground">
+                  {word}
+                </span>
+              );
+            }
+          })}
+        </div>
+      </div>
+      <div className="grid gap-2">
+        <Label htmlFor="amount">Amount</Label>
+        <Input
+          id="amount"
+          type="number"
+          value={amount}
+          placeholder="Amount in Satoshi..."
+          onChange={(e) => {
+            setAmount(e.target.value.trim());
+          }}
+          min={isSwap ? 50000 : 1}
+          required
+          autoFocus
+          className="[appearance:textfield] [&::-webkit-outer-spin-button]:appearance-none [&::-webkit-inner-spin-button]:appearance-none"
+          endAdornment={
+            <FormattedFiatAmount amount={Number(amount)} className="mr-2" />
+          }
+        />
+        <div className="grid">
+          <div className="flex justify-between text-muted-foreground text-xs sensitive slashed-zero">
+            <div>
+              {isSwap ? "Spending Balance: " : "On-chain Balance: "}
```

Comment:
We have a lot of conditionals e.g. `isSwap`

Can we split the two into different components?